format/ComputeIR.fbs

```diff
@@ -0,0 +1,241 @@
+  ///   "filter"
+  ///   "project"
+  ///   "aggregate"
+  ///   "join"
+  ///   "order_by"
+  ///   "limit"
+  ///   "common"
+  ///   "union"
+  ///   "literal"
+  ///   "interactive_output"
+  relation_name: string (required);
+
+  /// Parameters for `relation_name`; content/format may be unique to each
+  /// value of `relation_name`.
+  options: Buffer;
+
+  /// The arguments passed to `relation_name`.
+  arguments: [Relation] (required);
+
+  /// The schema of rows in this Relation
+  schema: Schema;
+}
+
+/// The contents of Relation.options will be FilterOptions
+/// if Relation.relation_name = "filter"
+table FilterOptions {
+  /// The expression which will be evaluated against input rows
+  /// to determine whether they should be excluded from the
+  /// "filter" relation's output.
+  filter_expression: Expression (required);
+}
+
+/// The contents of Relation.options will be ProjectOptions
+/// if Relation.relation_name = "project"
+table ProjectOptions {
+  /// Expressions which will be evaluated to produce to
+  /// the rows of the "project" relation's output.
+  expressions: [Expression] (required);
+}
+
+/// The contents of Relation.options will be AggregateOptions
+/// if Relation.relation_name = "aggregate"
+table AggregateOptions {
+  /// Expressions which will be evaluated to produce to
+  /// the rows of the "aggregate" relation's output.
+  aggregations: [Expression] (required);
+  /// Keys by which `aggregations` will be grouped.
+  keys: [Expression];
+}
+
+/// The contents of Relation.options will be JoinOptions
+/// if Relation.relation_name = "join"
+table JoinOptions {
+  /// The expression which will be evaluated against rows from each
+  /// input to determine whether they should be included in the
+  /// "join" relation's output.
+  on_expression: Expression (required);
+  join_kind: string;
+}
+
+/// Whether lesser values should precede greater or vice versa,
+/// also whether nulls should preced or follow values.
+enum Ordering : uint8 {
+  ASCENDING_THEN_NULLS,
+  DESCENDING_THEN_NULLS,
+  NULLS_THEN_ASCENDING,
+  NULLS_THEN_DESCENDING
+}
+
+table SortKey {
+  value: Expression (required);
+  ordering: Ordering = ASCENDING_THEN_NULLS;
+}
+
+/// The contents of Relation.options will be OrderByOptions
+/// if Relation.relation_name = "order_by"
+table OrderByOptions {
+  /// Define sort order for rows of output.
+  /// Keys with higher precedence are ordered ahead of other keys.
+  keys: [SortKey] (required);
+}
+
+/// The contents of Relation.options will be LimitOptions
+/// if Relation.relation_name = "limit"
+table LimitOptions {
+  /// Set the maximum number of rows of output.
+  count: long;
+}
+
+/// The contents of Relation.options will be CommonOptions
+/// if Relation.relation_name = "common"
+table CommonOptions {
+  /// Commons (CTEs in SQL) allow assigning a name to a stream
+  /// of data and reusing it, potentially multiple times and
+  /// potentially recursively.
+  name: string;
```

Comment:
Should this be `(required)`? What does  it mean for a CTE or alias to lack a name?

Reply:
This should indeed be required.
```suggestion
  name: string (required);
```